games-misc/an-anime-game-launcher-bin/an-anime-game-launcher-bin-2.3.8-r2.ebuild

```diff
@@ -39,25 +38,48 @@ src_unpack() {
 src_prepare(){
 	chmod +x ${P}.AppImage
 	./${P}.AppImage --appimage-extract || die "Extraction Failed"
-	chrpath -d "squashfs-root/public/discord-rpc/discord-rpc" || die "Patching Library Failed"
 	default
+
 	mv "squashfs-root/public/icons/256x256.png" "${PN}.png" || die
 	mv "squashfs-root/AppRun" "${PN}" || die
 	mv "squashfs-root/an-anime-game-launcher.desktop" "${PN}.desktop" || die
 }
 
 src_install(){
+
 	insinto "/usr/lib/${PN}"
 	doins "squashfs-root/resources.neu"
 	exeinto "/usr/lib/${PN}"
 	doexe "squashfs-root/an-anime-game-launcher"
-	doins -r "squashfs-root/public"
+
+	insinto "/usr/lib/${PN}/public" # specify path for public folders
+
+	#copy all files/folders from squashfs-root/public
+	doins -r "squashfs-root/public/discord-rpc"
+	doins -r "squashfs-root/public/dxvks.yaml"
+	doins -r "squashfs-root/public/hdiffpatch"
+	doins -r "squashfs-root/public/icons"
+	doins -r "squashfs-root/public/images"
+	doins -r "squashfs-root/public/locales"
+	doins -r "squashfs-root/public/neutralino.js"
+	doins -r "squashfs-root/public/runners.yaml"
+	doins -r "squashfs-root/public/shaders"
+
+	#create pix
 	insinto "/usr/share/pixmaps"
 	doins "${PN}.png"
+	#create bin
 	exeinto "/usr/bin"
 	doexe "${PN}"
+	#create desktop entry
 	insinto "/usr/share/applications/"
 	doins "${PN}.desktop"
+
+	cd ${FILESDIR}
+	insinto "/usr/lib/${PN}/public/discord-rpc/" # set path for discord-rpc binary
+	doins "discord-rpc"  # install discord-rpc to path
+
+	fperms 755 "/usr/lib/${PN}/public/discord-rpc/discord-rpc" # set permissions for discord-rpc
```

Comment:
if you use `exeinto` + `doexe` the file gets 755 permissions automatically

Reply:
>if you use exeinto + doexe the file gets 755 permissions automatically
Was not working, had tried that, binary didnt have execution perms for some reason

Reply:
Yes, I understand that. The install functions of the ebuilds often act a bit weird. 
I will test this out later. After all this is working with the binary file of the launcher.